### ISSUE TITLE
Eliminate All Compiler Warnings In Solution

### ISSUE THREAD

**Issue by mark-monteiro** [jellyfin/jellyfin#2149]
This is a general ticket to track progress on eliminating all warnings in the solution source code, including enabling missing analyzers and converting warnings to errors in release builds. @Bond-009, as I understand, you have already been working on this for a while. I thought I would start helping out with this task and we (and anyone else) can coordinate progress here.

The table below summarizes the progress on this task at the current HEAD commit on master:

|Project|Warnings|FxCop Analyzer|StyleCop Analyzer|Serilog Analyzer|Multithreading Analyzer|Warnings As Errors|
|-------|--------|--------------|-----------------|----------------|-----------------------|------------------|
|DvdLib|                               <ul><li>[x] </li></ul>|               <ul><li>[ ] </li></ul>|<ul><li>[ ] </li></ul>|<ul><li>[ ] </li></ul>|<ul><li>[ ] </li></ul>|<ul><li>[x] </li></ul>
|Emby.Dlna|                            <ul><li>[x] </li></ul>| <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Emby.Drawing|                         <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Emby.Naming|                          <ul><li>[x] </li></ul>|               <ul><li>[ ] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Emby.Notifications|                   <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Emby.Photos|                          <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Emby.Server.Implementations|          <ul><li>[ ] (644 Warnings)</li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[ ] </li></ul>
|Jellyfin.Api|                         <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Data|                        <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Drawing.Skia|                <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Server|                      <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Server.Implementations|      <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|MediaBrowser.Common|                  <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|MediaBrowser.Controller|              <ul><li>[ ] (1053 Warnings)</li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[ ] </li></ul>
|MediaBrowser.LocalMetadata|           <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|MediaBrowser.MediaEncoding|           <ul><li>[x] </li></ul>|  <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|MediaBrowser.Model|                   <ul><li>[ ] (2132 Warnings)</li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[ ] </li></ul>
|MediaBrowser.Providers|               <ul><li>[ ] (538 Warnings)</li></ul>| <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[ ] </li></ul>
|MediaBrowser.XbmcMetadata|            <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|RSSDP|                                <ul><li>[x] </li></ul>|               <ul><li>[ ] </li></ul>|<ul><li>[ ] </li></ul>|<ul><li>[ ] </li></ul>|<ul><li>[ ] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Api.Tests|                   <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Common.Tests|                <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Controller.Tests|            <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.MediaEncoding.Tests|         <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Model.Tests|                 <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Naming.Tests|                <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Jellyfin.Server.Implementations.Tests|<ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|MediaBrowser.Api.Tests|               <ul><li>[x] </li></ul>|               <ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>|<ul><li>[x] </li></ul>
|Total|4367 Warnings

**Comment by jvoisin** [comment]
Would it be possible to add [LGTM.com]( https://lgtm.com/projects/g/jellyfin/jellyfin?mode=list ) to the list? It found a lot of issues.

**Comment by dcparga** [comment]
Hey guys, I'm just wondering. I'm seeing 4028 warnings when building the solution. Some of the warnings are like "don't use space after new keyword", or others like to move types to their own file. I assume any warning fixed will be welcome, right?

**Comment by dkanada** [comment]
You're correct, anything that shows a warning is fair game for a pull request.

**Comment by Christian-Oleson** [comment]
I'm willing to help out with this if you are looking for help

**Comment by jbatt33** [comment]
I just built Jellyfin and I'm not seeing any warnings. Is this bug fixed or am I looking at the wrong repo. Maybe I need to build harder.

**Comment by Bond-009** [comment]
@jbatt33 are you building in debug mode?

**Comment by jbatt33** [comment]
Yes, I did both debug and release builds of the jellyfin under VS 2022 and did not get any warnings or errors, but get 3 messages: 
![image](https://user-images.githubusercontent.com/41477628/228947738-1b5a13eb-1ea8-45b7-ae05-ad1a9bad2438.png)

**Comment by jbatt33** [comment]
Thanks, I deleted that line from the project file and it's working as expected.  I'm going to see if I can tackle any of these.

**Comment by LoremFooBar** [comment]
I'd like to try to clear some of these warnings. I already started with the simpler ones. Some warnings require refactoring that is easy to do, but hard to determine whether they are breaking changes. I'd like to try to tackle these as well later on.

**Comment by gsedubun** [comment]
I'd like to try to clear some of these.

**Comment by Crystora** [comment]
Hi @mark-monteiro @Bond-009, I’d be happy to take this on if it still needs work. Please let me know. Thanks!

**Comment by felix920506** [comment]
@bittoby you can easily confirm the current status by compiling the code and see what warnings popup, then work on some of those warnings and send a PR.
